### PR TITLE
[refactor] Extract plot utils into dedicated file

### PR DESCRIPTION
**Pull request by LeonieFreisinger**
**Problem** referring  #965 
Helper functions in plotting related files are defined in actual plotting files. This enlarges the file content.

**Solution**
Relocate plotting-related helper functions to a separate file `utils_plot.py`

Key Changes: 
- Check the plotting-related .py files for helper functions and move them to `utils_plot.py`
- Check utils.py for plotting-related helper functions and move them to `utils_plot.py`

Functions that were not relocated: 
`plot_trend_change(), plot_trend(), plot_daily(), plot_weekly(), plot_yearly(), plot_custom_season(), plot_lagged_weights(), plot_scalar_weights(), plot_forecast_component(), plot_multiforecast_component()`

### CODE REVIEW

**Comment by LeonieFreisinger** [comment]
@noxan would be amazing if you can review this PR.  Should be a fast one.
P.S: I did not relocate the functions that call a plot command, since they in my eyes are not a helper function. 

Thanks a lot!

File: neuralprophet/utils_plot.py [hunk 1]
```diff
@@ -0,0 +1,532 @@
...
+            "multiplicative_events": multiplicative_events,
+            "lagged_scalar_regressors": lagged_scalar_regressors,
+            "overwriting_unknown_data_normalization": overwriting_unknown_data_normalization,
+            "df_name": df_name,
+        }
+    elif validator == "plot_components":
+        valid_configuration = {
+            "components_list": plot_components,
+        }
+    return valid_configuration
```
Comment (noxan) [inline neuralprophet/utils_plot.py:532]: At some point we should definitely refactor this method... also in parts related to #1004

**Review by noxan — APPROVED** [review]
LGTM - one note on this massive function to build the component plotting configuration (it is too massive) for a follow up and just wondering: would `plot_utils.py` be a better name? Otherwise good to go from my side.

**Comment by noxan** [comment]
@LeonieFreisinger let me know what you think and otherwise I'd merge it directly (possibly create follow up issues).

**Comment by LeonieFreisinger** [comment]
> LGTM - one note on this massive function to build the component plotting configuration (it is too massive) for a follow up and just wondering: would `plot_utils.py` be a better name? Otherwise good to go from my side.

@noxan thanks for the fast review. Initially planned to name the module plot_utils.py. However, I aligned it with `utils.py `and `utils_torch.py`. I agree with changing it. Should we rename the `utils_torch.py` as well to `torch_utils.py`? 
Then I open a new issue for it to keep it separate. As well as for the `get_valid_configuration()` and link it to #1004.

**Comment by noxan** [comment]
Thanks @LeonieFreisinger - good point about having consistent naming for our `util*` files, let's take the discussion to the issue you created :)

**Comment by ourownstory** [comment]
Congratulations @LeonieFreisinger you merged our #1000 th Pull Request! You won a free tour around Stanford campus! Please let me know your availability.

**Comment by LeonieFreisinger** [comment]
> Congratulations @LeonieFreisinger you merged our #1000 th Pull Request! You won a free tour around Stanford campus! Please let me know your availability.

definitely feel honored of saving PR1000. hahhaha @ourownstory  almost as good as winning the lottery.

**Pull request merged**